File: 字符串/day2/翻转字符串中的单词/reversw_words_in_string_pfan8.cpp
class Solution {
public:
    string reverseWords(string s) {
        // 时间复杂度 O(n)
        // 空间复杂度 O(n)
        string res = "";
        int pos = 0;
        for(int i = 0; i < s.length(); i++) {
            if(s[i] != ' ') {
                pos = i;
                while(++i < s.length() && s[i] != ' ') {}
                res = s.substr(pos,i-pos) + " " + res;
            }
        }
        return res.substr(0,res.length()-1);
    }
};

File: 字符串/翻转字符串中的单词/reversw_words_in_string_pfan8.java
class Solution {
    public String reverseWords(String s) {
        // 时间复杂度O(n)
        // 空间复杂度O(n)
        if (s == null || s.length() == 0) {
            return "";
        }
        String[] arr = s.split(" ");
        StringBuilder res = new StringBuilder();
        
        for (int i = arr.length - 1; i >= 0; i--) {
            String word = arr[i];
            
            if (!word.equals("")) {
                if (res.length() > 0) {
                    res.append(" ");
                }
                res.append(word);
            } 
        }
        return res.toString();
    }
}